# hvPlot Widgets

Your parents need help identifying the best states to move to that have the lowest Diabetes care costs. Use hvPlot and its widgets to visualize hospital claims data in order to identify potential states for recommendation.

In [10]:
import pandas as pd
import hvplot.pandas
from pathlib import Path

### Step 1: Prep Data

In [11]:
# Read in hospital claims data
hospital_data = pd.read_csv(
    Path("../Resources/hospital_claims.csv"), 
    header=0
)

# View the DataFame
hospital_data.head()

DRG Definition  Provider Id  \
0  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10001   
1  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10005   
2  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10006   
3  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10011   
4  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC        10016   

                      Provider Name     Provider Street Address Provider City  \
0  SOUTHEAST ALABAMA MEDICAL CENTER      1108 ROSS CLARK CIRCLE        DOTHAN   
1     MARSHALL MEDICAL CENTER SOUTH  2505 U S HIGHWAY 431 NORTH          BOAZ   
2    ELIZA COFFEE MEMORIAL HOSPITAL          205 MARENGO STREET      FLORENCE   
3                 ST VINCENT'S EAST  50 MEDICAL PARK EAST DRIVE    BIRMINGHAM   
4     SHELBY BAPTIST MEDICAL CENTER     1000 FIRST STREET NORTH     ALABASTER   

  Provider State  Provider Zip Code Hospital Referral Region Description  \
0             AL              36301                          AL - Dothan   
1             AL              35957                      AL - Birmingham   
2             AL              35631                      AL - Birmingham   
3             AL              35235                      AL - Birmingham   
4             AL              35007                      AL - Birmingham   

   Total Discharges  Average Covered Charges  Average Total Payments  \
0                91                 32963.07                 5777.24   
1                14                 15131.85                 5787.57   
2                24                 37560.37                 5434.95   
3                25                 13998.28                 5417.56   
4                18                 31633.27                 5658.33   

   Average Medicare Payments  
0                    4763.73  
1                    4976.71  
2                    4453.79  
3                    4129.16  
4                    4851.44

In [12]:
# Slice the data for the DRG Definition "638 - DIABETES W CC"
procedure_638_charges = hospital_data[
    hospital_data["DRG Definition"] == "638 - DIABETES W CC"
]

# View the DataFrame
procedure_638_charges.head()

DRG Definition  Provider Id                     Provider Name  \
125166  638 - DIABETES W CC       330004                 KINGSTON HOSPITAL   
127575  638 - DIABETES W CC        10001  SOUTHEAST ALABAMA MEDICAL CENTER   
127576  638 - DIABETES W CC        10005     MARSHALL MEDICAL CENTER SOUTH   
127577  638 - DIABETES W CC        10006    ELIZA COFFEE MEMORIAL HOSPITAL   
127578  638 - DIABETES W CC        10011                 ST VINCENT'S EAST   

           Provider Street Address Provider City Provider State  \
125166                396 BROADWAY      KINGSTON             NY   
127575      1108 ROSS CLARK CIRCLE        DOTHAN             AL   
127576  2505 U S HIGHWAY 431 NORTH          BOAZ             AL   
127577          205 MARENGO STREET      FLORENCE             AL   
127578  50 MEDICAL PARK EAST DRIVE    BIRMINGHAM             AL   

        Provider Zip Code Hospital Referral Region Description  \
125166              12401                          NY - Albany   
127575              36301                          AL - Dothan   
127576              35957                      AL - Birmingham   
127577              35631                      AL - Birmingham   
127578              35235                      AL - Birmingham   

        Total Discharges  Average Covered Charges  Average Total Payments  \
125166                21                 20006.57                 6048.85   
127575                32                 21175.81                 4678.43   
127576                12                  9719.16                 4863.75   
127577                35                 17021.54                 4434.57   
127578                14                 15875.50                 5176.07   

        Average Medicare Payments  
125166                    5177.85  
127575                    4047.68  
127576                    4203.41  
127577                    3537.20  
127578                    3394.64

In [13]:
# Select data just for NJ data to get a baseline for Diabetes costs in New Jersery only
procedure_NJ_charges = procedure_638_charges.loc[
    hospital_data["Provider State"] == "NJ"
]

# View the DataFrame
procedure_NJ_charges.head()

DRG Definition  Provider Id  \
128582  638 - DIABETES W CC       310001   
128583  638 - DIABETES W CC       310002   
128584  638 - DIABETES W CC       310003   
128585  638 - DIABETES W CC       310005   
128586  638 - DIABETES W CC       310006   

                                            Provider Name  \
128582               HACKENSACK UNIVERSITY MEDICAL CENTER   
128583                  NEWARK BETH ISRAEL MEDICAL CENTER   
128584  PALISADES MEDICAL CTR-NY PRESBYTERIAN HEALTHCA...   
128585                           HUNTERDON MEDICAL CENTER   
128586                        ST MARY'S HOSPITAL, PASSAIC   

       Provider Street Address Provider City Provider State  \
128582         30 PROSPECT AVE    HACKENSACK             NJ   
128583           201 LYONS AVE        NEWARK             NJ   
128584           7600 RIVER RD  NORTH BERGEN             NJ   
128585      2100 WESCOTT DRIVE    FLEMINGTON             NJ   
128586           350 BOULEVARD       PASSAIC             NJ   

        Provider Zip Code Hospital Referral Region Description  \
128582               7601                      NJ - Hackensack   
128583               7112                          NJ - Newark   
128584               7047                      NJ - Hackensack   
128585               8822                    PA - Philadelphia   
128586               7055                      NJ - Hackensack   

        Total Discharges  Average Covered Charges  Average Total Payments  \
128582                43                 45439.81                 6579.39   
128583                60                 61179.28                 9840.78   
128584                29                 43669.75                 7099.65   
128585                12                 17753.91                 6081.91   
128586                17                 36770.88                 6574.05   

        Average Medicare Payments  
128582                    5873.06  
128583                    8989.73  
128584                    6268.68  
128585                    4423.58  
128586                    5843.47

In [14]:
# Using the NJ DataFrame, create a DataFrame consisting of 
# Average Medicare Payments and Provider ID columns
NJ_average_total_payments = procedure_NJ_charges[
    ["Average Total Payments", "Provider Id"]
]

# View the DataFrame
NJ_average_total_payments.head()

Average Total Payments  Provider Id
128582                 6579.39       310001
128583                 9840.78       310002
128584                 7099.65       310003
128585                 6081.91       310005
128586                 6574.05       310006

In [15]:
# The sum of NJ's average total payments
NJ_average_total_payments['Average Total Payments'].sum()

379875.32

### Step 2: Calculate the Average Total Payments by Provider State

In [17]:
# Using the procedure_638_charges DataFrame, select state and average total payments from your data.

payment_by_state = procedure_638_charges[["Average Total Payments", "Provider State"]]

# Aggregate the sum of payments by Provider State
total_payment_by_state = payment_by_state.groupby("Provider State").sum()

# Visualize the DataFrame
total_payment_by_state.head()

Average Total Payments
Provider State                        
AK                            15525.04
AL                           194535.26
AR                           111746.03
AZ                           158308.98
CA                          1201241.89

### Step 3: Plot the payment by state date in a bar chart

In [20]:
# Visualize the data using an hvPlot bar chart
total_payment_by_state.hvplot.bar(
    rot=90
).opts(
    yformatter='%.0f'
)

:Bars   [Provider State]   (Average Total Payments)

**Question:** How does the total cost of diabetes payments in New Jersey compare to that of surrounding states like NY, CT, DE, and PA?

**Answer:** The cost of diabetes treatments in NJ are higher than some of the surrounding states but lower than others. Total payments in NJ sum to 379,875. Of the surrounding states,the cost is lowest in DE at 29,633. The cost is highest in NY at 871,073. PA comes in at 490,962. CT at 131,976.

### Step 4: Sort the cost data from low to high and then plot a bar chart


In [22]:
# Sort the total payment by state data by Average Total Payments
# and visualize the values in a bar chart
total_payment_by_state.sort_values("Average Total Payments").hvplot.bar(
    rot=90
).opts(
    yformatter='%.0f'
)


:Bars   [Provider State]   (Average Total Payments)

**Question:** List the 10 states with the lowest total payments for diabetes care.

**Answer:** The 10 states are: WY, HI, AK, SD, VT, MT, ND, ID, DE, and NE

**Question:** How many states have a total diabetes payments of less than 200,000?

**Answer:** There are 33 states with total diabetes payments of less than 200,000.

**Question:** What is the approximate difference in the total payments for diabetes payments between the state with the 10th highest payments and the state with the highest payments?

**Answer:** The state of CA has the highest total payments at 1,201,241. The state with the 10th higest payments is MI at 345,449. The difference in payments between those two states is 855,792.